### PR TITLE
ANTLRv4 indent and code snippet support.

### PR DESCRIPTION
**Pull request by lkishalmi**
This is my last round of features for ANTLR support, mainly for v4 grammars. Add indent support in rule definitions, and a few code shortcuts. 

Beside of that, prevented a few of NPE-s that could happen on parsing an incomplete grammar. Also removed the ConsoleErrorListener.

### CODE REVIEW

File: java/languages.antlr/src/org/netbeans/modules/languages/antlr/v4/Antlr4CompletionProvider.java [hunk 1]
```diff
@@ -143,12 +129,12 @@ protected void query(CompletionResultSet resultSet, Document doc, int caretOffse
...
                     } else {
-                        //Empty grammar so far offer lexer and grammar
-                        addTokens("", caretOffset, resultSet, "lexer", "grammar");
+                        //Empty grammar so far offer lexer, parser and grammar
+                        addTokens("", caretOffset, resultSet, "lexer", "parser", "grammar");
                     }
                 }
             } catch (Throwable th) {
-                System.out.println(th);
+                //System.out.println(th);
```
Comment (mbien) [inline java/languages.antlr/src/org/netbeans/modules/languages/antlr/v4/Antlr4CompletionProvider.java:137]: was this intended?  Would be better if it wouldn't be `Throwable`, It is always a bit dangerous to catch that without handling the edge cases.

File: java/languages.antlr/src/org/netbeans/modules/languages/antlr/v4/Antlr4Formatter.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+    @Override
+    public boolean needsParserResult() {
+        return false;
+    }
+
+    @Override
+    public int indentSize() {
+        return 3;
+    }
```
Comment (mbien) [inline java/languages.antlr/src/org/netbeans/modules/languages/antlr/v4/Antlr4Formatter.java:138]: 3 space indent is evil since it looks almost right :)

**Review by mbien — APPROVED** [review]
looks good as far as I can tell. But I didn't "think through" the formatting code. Its been a while since I wrote an Antlr grammar.

This is also fairly low risk for inclusion. +1 for merging this before feature freeze.

File: java/languages.antlr/src/org/netbeans/modules/languages/antlr/v4/Antlr4Formatter.java (hunk 2 above)
Comment (lkishalmi) [reply java/languages.antlr/src/org/netbeans/modules/languages/antlr/v4/Antlr4Formatter.java]: Just checked a few grammars today randomly. Most of them were using 4 spaces, but a few of them were using 3. It seems our ANTLR v3 and v4 grammars using 3 spaces are kind of a minority.

File: java/languages.antlr/src/org/netbeans/modules/languages/antlr/v4/Antlr4CompletionProvider.java (hunk 1 above)
Comment (lkishalmi) [reply java/languages.antlr/src/org/netbeans/modules/languages/antlr/v4/Antlr4CompletionProvider.java]: Well, removed that one.

**Comment by lkishalmi** [comment]
Well, a formatter code is never nice, as far as I checked. The basic idea here is using a heuristic based on the Lexer to determine whether we are in a rule or not. While in rule, the whitespace is examined for each line break, and indent the line according to the inrule state.

**Pull request merged**